dsp/modules/gpt3.py

```diff
@@ -66,8 +68,40 @@ def __init__(
     ):
         super().__init__(model)
         self.provider = "openai"
-        openai.api_type = api_provider
 
+        # Assert that all variables are available
+        assert api_key is not None, "Must provide api_key for OpenAI/Azure API"
```

Comment:
The key is not generally necessary, it can be obtained from the ENV variables

Reply:
it depends on the authentication type used; Azure offers two ways; one using the env variable, and another one which uses rotating keys (using MS Azure services)